qt_ui/windows/basemenu/QBaseMenu2.py

```diff
@@ -126,6 +127,9 @@ def cheat_capture(self) -> None:
         self.game_model.game.initialize_turn(events)
         EventStream.put_nowait(events)
         GameUpdateSignal.get_instance().updateGame(self.game_model.game)
+        state = self.game_model.game.check_win_loss()
```

Comment:
Why is this being duplicated? Shouldn't this all be handled in a single handler that responds to `updateGame`?

Reply:
Refactored to `game.process_win_loss`

Reply:
I've been at it for some time now but I keep hitting a brick wall. What I had assured the signal would only be triggered once. With your suggestions I can't seem to guarantee that...

Reply:
Changing this:
```
GameUpdateSignal.get_instance().updateGame(self.game_model.game)
state = self.game_model.game.check_win_loss()
if state == TurnState.WIN or state == TurnState.LOSS:
    GameUpdateSignal.get_instance().endGame(state)
```
into:
```
self.close()
```

Would work for cheat-capture given we refactor the signal-trigger to `QLiberationWindow.setGame`. As for normal capture events I'm still having a double pop-up...

Reply:
Well after closer inspection it's not the game that gets released twice, but rather an update that gets sent to the front-end, which is exactly what I don't want as this wipes the map. I'm giving it one last try to refactor these signal triggers but if that doesn't work out, I'm recommending leaving the signal triggers where they are...